x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/trigger/schedule/tool/CronEvalTool.java

```diff
@@ -57,18 +58,25 @@ void execute(Terminal terminal, String expression, int count) throws Exception {
 
         DateTime date = DateTime.now(DateTimeZone.UTC);
         terminal.println("Now is [" + formatter.print(date) + "]");
-        terminal.println("Here are the next " + count + " times this cron expression will trigger:");
 
         Cron cron = new Cron(expression);
         long time = date.getMillis();
-        for (int i = 0; i < count; i++) {
+        int i = 0;
+        List<String> times = new ArrayList<>();
+
+        for (; i < count; i++) {
             long prevTime = time;
             time = cron.getNextValidTimeAfter(time);
             if (time < 0) {
-                throw new UserException(ExitCodes.OK, (i + 1) + ".\t Could not compute future times since ["
-                    + formatter.print(prevTime) + "] " + "(perhaps the cron expression only points to times in the past?)");
+                if (i == 0) {
+                    throw new UserException(ExitCodes.OK, "Could not compute future times since ["
+                            + formatter.print(prevTime) + "] " + "(perhaps the cron expression only points to times in the past?)");
+                }
+                break;
             }
-            terminal.println((i+1) + ".\t" + formatter.print(time));
+            times.add((i + 1) + ".\t" + formatter.print(time));
         }
+        terminal.println((i == count ? "Here are the next " : "There are ") + i + " times this cron expression will trigger:");
```

Comment:
I don't think we need this special handling here (introducing state that requires following by filling up the list). Just adding the additional if-statement plus break should be sufficient.

Reply:
I made the change as originally the following results were printed by the cli tool:
`0 0 0 1-6 * ?` resulted in
```
Valid!
Now is [Wed, 15 Aug 2018 11:28:34]
Here are the next 10 times this cron expression will trigger:
1.	Sat, 1 Sep 2018 02:00:00
2.	Sun, 2 Sep 2018 02:00:00
3.	Mon, 3 Sep 2018 02:00:00
4.	Tue, 4 Sep 2018 02:00:00
5.	Wed, 5 Sep 2018 02:00:00
6.	Thu, 6 Sep 2018 02:00:00
7.	Mon, 1 Oct 2018 02:00:00
8.	Tue, 2 Oct 2018 02:00:00
9.	Wed, 3 Oct 2018 02:00:00
10.	Thu, 4 Oct 2018 02:00:00
```
And a one time cron, such as `0 3 23 8 9 ? 2019` resulted in
```
Valid!
Now is [Wed, 15 Aug 2018 11:29:37]
Here are the next 10 times this cron expression will trigger:
1.	Mon, 9 Sep 2019 01:03:00
```
I found it is a little bit confusing that in this case the message was `Here are the next 10 time` and only a single one was printed, hence I tried to clarify the message.

I will revert the change.